enhancements/ingress/haproxy-router-compression.md

```diff
@@ -0,0 +1,363 @@
+but these are not in scope of this proposal.
+
+### Risks and Mitigations
+
+Overuse of compression can cause the HAProxy to devote more resources to compression than to other operations,
+compromising its operation.  We do not have any evidence that this will occur, and give detailed notes on use of
+necessary MIME types to mitigate pointless compression and overuse of compression.  Should this become a problem
+in the future, we can add default values for the process-level directives mentioned above (`maxcomprate`,
+`maxcompcpuusage`, and `tune.comp.maxlevel`).  The team consensus is that adding more tunable knobs at this stage is
+premature and could cause more issues than it mitigates.
+
+We considered disallowing wildcard MIME types (e.g. `application/*`), so that explicit choices must be made on which
+MIME types are subject to compression, but as mentioned, it is more likely that any MIME type omission would generate
+a new RFE than for there to be a support issue caused by excessive pointless compression.
+
+Security and UX are not a part of this proposal, but API team members will be invited to comment.
+
+## Design Details
+Design details are covered in the Proposal section, but some additional detailed notes from the
+[HAProxy compression documentation](https://www.haproxy.com/documentation/hapee/latest/load-balancing/compression/)
+are listed below.
+
+### Compression types
+MIME types to apply compression to,  are read from the Content-Type response header field. If a MIME type is not set,
+HAProxy tries to compress all responses that are not compressed by the server.  Therefore, at least one MIME type is
+required in the `HTTPCompressionPolicy`.  We consider the fact that cluster admins may want to create and use their
+own MIME type, but the allowed set of characters is defined in a kubebuilder validation
+pattern  of `"[a-zA-Z0-9\\+-;= ]+/([a-zA-Z0-9\\\+-;= ]+|\\*)"`, to allow a token of alphanumeric plus "_-;= " followed
+by "/" followed by another alphanumeric token of the same pattern, or "*", the wildcard character.
+
+### Open Questions
+n/a
+
+### Test Plan
+- Unit tests will validate
+  - that at least one HTTP compression MIME type exists and any listed MIME types adhere to the
+  correct pattern
+  - that the HTTP compression directive appears in the `defaults` section of the HAProxy template
+  when it is enabled and properly configured on the IngressController spec
+  - that a downgrade to version 4.9 (or version previous to feature implementation version) will not retain the
+  compression configuration of 4.10 (the environment variables should no longer be exposed)
+- E2e tests will validate
+  - that the IngressController spec accepts and enables compression in the `defaults` section of
+  the generated haproxy.cfg
+  - that only requested MIME types receive compression
+
+Note that HAProxy Enterprise **disables** HTTP compression when:
```

Comment:
We use HAProxy (not HAProxy Enterprise), but I think both have the same behavior.